runtime/webhookserver/webhooktunnel.go

```diff
@@ -0,0 +1,100 @@
+package webhookserver
+
+import (
+	"errors"
+	"net/http"
+	"sync"
+
+	"github.com/taskcluster/slugid-go/slugid"
+	"github.com/taskcluster/taskcluster-client-go"
+	"github.com/taskcluster/taskcluster-client-go/auth"
+	"github.com/taskcluster/webhooktunnel/util"
+	"github.com/taskcluster/webhooktunnel/whclient"
+)
+
+// WebhookTunnel wraps a whclient instance as a WebHookServer
+type WebhookTunnel struct {
+	m        sync.RWMutex
+	handlers map[string]http.Handler
+	// url is assigned when the client connects to the proxy
+	client *whclient.Client
+}
+
+// NewWebhookTunnel returns a pointer to a new WebhookTunnel instance
+func NewWebhookTunnel(credentials *tcclient.Credentials) (*WebhookTunnel, error) {
+	configurer := func() (whclient.Config, error) {
+		authClient := auth.New(credentials)
+		whresp, err := authClient.WebhooktunnelToken()
+		if err != nil {
+			return whclient.Config{}, errors.New("could not get token from tc-auth")
+		}
+
+		// This hack is needed since proxyURL in auth config is set as a http url
+		proxyURL := whresp.ProxyURL
+		proxyURL = util.MakeWsURL(proxyURL)
+
+		return whclient.Config{
+			ID:        whresp.TunnelID,
+			ProxyAddr: proxyURL,
+			Token:     whresp.Token,
+			UseDomain: true,
```

Comment:
Why do we need to pass this? Isn't it evident from the URL we get back from registration... and isn't the registration URL `ProxyAddr`?